entity_mongo_model.go

```diff
@@ -73,13 +74,12 @@ type entityMongoModel[T any] struct {
 }
 
 // NewEntityMongoModel returns a new instance of EntityMongoModel for the provided model type and options.
-func NewEntityMongoModel[T any](modelType T, opts EntityMongoOptions) (EntityMongoModel[T], error) {
-	dbConnection := opts.dbConnection
-	if dbConnection == nil {
+func NewEntityMongoModel[T any](modelType T, schemaOpts schemaopt.SchemaOptions) (EntityMongoModel[T], error) {
```

Comment:
Add a note in documentation that we need to set the db connection at the start of the app, if not already mentioned

Reply:
done already.